src/components/AudioLevelIndicator/AudioLevelIndicator.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef } from 'react';
+import { AudioTrack, LocalAudioTrack, RemoteAudioTrack } from 'twilio-video';
+import MicOff from '@material-ui/icons/MicOff';
+import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
+
+let clipId = 0;
+const getUniqueClipId = () => clipId++;
+
+// @ts-ignore
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+let audioContext: AudioContext;
+
+function AudioLevelIndicator({
+  size,
+  audioTrack,
+  background,
+}: {
+  size?: number;
+  audioTrack?: AudioTrack;
+  background?: string;
+}) {
+  const SIZE = size || 24;
+  const ref = useRef<SVGRectElement>(null);
+  const isTrackEnabled = useIsTrackEnabled(audioTrack as LocalAudioTrack | RemoteAudioTrack);
+
+  useEffect(() => {
+    const SVGClipElement = ref.current;
+    let animationFrameID: number;
+
+    if (audioTrack && isTrackEnabled && SVGClipElement) {
+      audioContext = audioContext || new AudioContext();
+      const stream = new MediaStream();
+      stream.addTrack(audioTrack.mediaStreamTrack.clone());
+      const audioSource = audioContext.createMediaStreamSource(stream);
+
+      const analyser = audioContext.createAnalyser();
```

Comment:
Have we tested this on other browsers? I know there are audio level issues on safari, especially if the browser goes from focus -> unfocus -> focus. When this happens, you will get zero levels.

Reply:
@charliesantos This is the whole point of the indicator. When this happens, the zero levels should result in the app displaying a warning asking the customer to re-join the room.

Reply:
@manjeshbhargav I understand the goal (I think). However, the issue I'm seeing with this approach is, we are using volume levels to determine whether the customer should re-join the room or not. Getting volume levels in this manner proved to have issues on Safari and not reliable. Here's an example.

I use safari, join a room. Everything is fine. My microphone works and everyone can hear me. Audio level indicator looks good also. Then, zero level happens (you can trigger by going out of focus on safari), but my mic still works and everyone can still hear me. Would we still show the warning and ask the customer to re-join the room?

Also, I have not tested the react app with this on safari. But I have tested on VoiceJS SDK because we currently have the same implementation.

Ultimately, the point I'm trying to make is, we should not rely on this across browser or if we do, since this might be the only option we have, we should be able to detect when these zero level issue happens and handle it instead of always showing the warning.

Reply:
I just tested on Safari and I was able to reproduce the issue where zero levels happen.
Steps:
1. Load app on safari. Observe the audio indicator. Should work when you speak.
2. Open a new tab and make sure the new tab is focused
3. Now click the original tab on step 1.
4. Try to speak, audio indicator stops working.

When this happens, I don't think we should just show a generic message asking the customer to re-join. We should either find a fix for this or detect when this happens so we can show a different message. I don't have a solution on how to detect the bug though.

Reply:
@charliesantos I pushed up a fix for this.